### PR TITLE
chore(monitoring): add new action group

### PR DESCRIPTION
**Pull request by Garma00**
### List of changes

* Add new io_com action group to alert related to shared resources;
* Replace generic error action group with io_com action group for iocom resources.

### Motivation and context

We want to isolate iocom related alert in a dedicated slack channel.

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Comment by Garma00** [comment]
@christian-calabrese @mamu0 Can you help me to understand why those resources are going to be destroyed here? https://github.com/pagopa/io-infra/actions/runs/10902600014/job/30254861123?pr=1164

**Comment by Garma00** [comment]
> > @christian-calabrese @mamu0 Can you help me to understand why those resources are going to be destroyed here? [pagopa/io-infra/actions/runs/10902600014/job/30254861123?pr=1164](https://github.com/pagopa/io-infra/actions/runs/10902600014/job/30254861123?pr=1164)
> 
> It's because some deps to the #1160 I try to merge it today

Ok thank you @Krusty93

File: src/core/data.tf [hunk 1]
```diff
@@ -169,6 +169,11 @@ resource "azurerm_monitor_metric_alert" "cosmos_api_throttling_alert" {
     webhook_properties = {}
   }
 
+  action {
+    action_group_id    = data.azurerm_monitor_action_group.io_com_action_group.id
+    webhook_properties = {}
+  }
```
Comment (lucacavallaro) [inline src/core/data.tf:175]: I would remove this `action`. 

IMHO we should define our own `azurerm_monitor_metric_alert` in https://github.com/pagopa/io-messages with the query configured to look only for our collections (`messages`, `messages-view`, `messages-status` ecc.)

**Review by lucacavallaro — CHANGES_REQUESTED** [review]
PR description needed 👀

File: src/core/data.tf (hunk 1 above)
Comment (Garma00) [reply src/core/data.tf]: https://github.com/pagopa/io-infra/pull/1164/commits/ef797ff60c9c4ced53a1be97d219e7f5261a4363

File: src/core/monitor.tf [hunk 2]
```diff
@@ -7,3 +7,8 @@ data "azurerm_monitor_action_group" "error_action_group" {
   name                = "${var.prefix}${var.env_short}error"
   resource_group_name = azurerm_resource_group.rg_common.name
 }
+
+data "azurerm_monitor_action_group" "io_com_action_group" {
+  name                = "io-p-com-error-ag-01"
+  resource_group_name = "io-p-itn-msgs-rg-01"
+}
```
Comment (lucacavallaro) [inline src/core/monitor.tf:14]: since this data is not longer used, it can be removed

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: src/core/monitor.tf (hunk 2 above)
Comment (Garma00) [reply src/core/monitor.tf]: https://github.com/pagopa/io-infra/pull/1164/commits/7b729bdb44e7bda346bb6ebcfd00a1f9a4839a0d

**Review by lucacavallaro — APPROVED** [review]

**Review by christian-calabrese — APPROVED** [review]

**Comment by Garma00** [comment]
@Krusty93 it seems we are having some permissions issues in the github action, the azure pipeline is fine

**Comment by mamu0** [comment]
> @Krusty93 it seems we are having some permissions issues in the github action, the azure pipeline is fine

As i can see this problem is born after 3.93 `azurerm` provider version, and the only two solution are
add the skip flang into the provider configuration (look at [this](https://github.com/hashicorp/terraform-provider-azurerm/issues/24804) discussion)
E.g.
```
provider "azurerm" {
  features {}
  skip_provider_registration = true
}
```
or register the provider, maybe with this [terraform resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_provider_registration).

**Comment by Garma00** [comment]
> As i can see this problem is born after 3.93 `azurerm` provider version, and the only two solution are add the skip flang into the provider configuration (look at [this](https://github.com/hashicorp/terraform-provider-azurerm/issues/24804) discussion) E.g.

@mamu0 It seems strange to me that this is related to the provider version because we are not having any trouble in the devops pipeline as you can see [here](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build/results?buildId=173229&view=logs&j=12882ec1-3675-5f0c-ffd4-78407b4125fa&t=b027e8fb-d587-5a33-a945-243166444099), also everything was fine before updating the branch and I can't see any closed PR where we updated the provider version. Are you sure that we are using v3.93 now?

**Comment by mamu0** [comment]
> > As i can see this problem is born after 3.93 `azurerm` provider version, and the only two solution are add the skip flang into the provider configuration (look at [this](https://github.com/hashicorp/terraform-provider-azurerm/issues/24804) discussion) E.g.
> 
> @mamu0 It seems strange to me that this is related to the provider version because we are not having any trouble in the devops pipeline as you can see [here](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build/results?buildId=173229&view=logs&j=12882ec1-3675-5f0c-ffd4-78407b4125fa&t=b027e8fb-d587-5a33-a945-243166444099), also everything was fine before updating the branch and I can't see any closed PR where we updated the provider version. Are you sure that we are using v3.93 now?

I checked more in detail and the error is this:
`...Errors were: Cannot register provider Microsoft.TimeSeriesInsights with Azure Resource Manager: unexpected status 400 with error: DisallowedProvider: The operation is not permitted for namespace 'Microsoft.TimeSeriesInsights'....`
And in fact the downloaded version of the provider is `3.84` because of the terraform-azure-v3 modules that are blocked at version `7.61` (now we are at `8.44`).

Trying to increase the version the problem does not occur, but there would be several changes and replace to evaluate.
@Krusty93 what do you say? Is this a planned change?

**Comment by Garma00** [comment]
> Shall we go ahead by updating all the required provider version?

Yess I'm going to @Krusty93

**Comment by Garma00** [comment]
Can you help me understand why the `terraform validate` fails here? I am not having any issues locally @mamu0

**Comment by mamu0** [comment]
> Can you help me understand why the `terraform validate` fails here? I am not having any issues locally @mamu0

Have you tried to delete `.terraform` folder and `.terraform.lock.hcl` locally? Because with them the `terraform init` command will download the old version

**Comment by Garma00** [comment]
> Have you tried to delete `.terraform` folder and `.terraform.lock.hcl` locally? Because with them the `terraform init` command will download the old version

Even after that, I still have no errors when running precommit locally but it fails in the gh action, do you have any other suggestion? cc @Krusty93

**Comment by Garma00** [comment]
closing in favour of #1213

**Pull request closed without merging**